scheduler/graphql_mid/schema.py

```diff
@@ -67,12 +61,22 @@ async def load_sources_files(self, files_input: UseFilesSourceInput) -> SourceFi
                                                  loaded=False,
                                                  msg='Handler not implemented yet!')
 
+    # @strawberry.mutation
+    # async def load_sources_form(self):
+    # This method loads basic conditions from a form similar to Mercury demo.
+
     @strawberry.mutation
-    def change_origin(new_origin: SOrigin) -> ChangeOriginSuccess:
-        old = str(builder.sources.origin)
+    def change_origin(self, new_origin: SOrigin, mode: SchedulerModes) -> ChangeOriginSuccess:
+
+        old = str(sources.origin)
+        new = str(new_origin)
+        if new == 'OCS' and mode is SchedulerModes.SIMULATION:
+            raise ValueError('Simulation mode can only work with GPP origin source.')
+        elif new == 'GPP' and mode is SchedulerModes.VALIDATION:
```

Comment:
It would be good to have the magic strings 'OCS' and 'GPP' defined elsewhere since they're not only used here but in asserts, but it's not the end of the world.